demos/asp-net-core/Divergent.Sales.API.Host/Controllers/OrdersController.cs

```diff
@@ -9,18 +9,17 @@ namespace Divergent.Sales.API.Host.Controllers
     [RoutePrefix("api/orders")]
     public class OrdersController : ApiController
     {
-        [HttpGet]
-        public dynamic Get(int id)
+        [HttpGet, Route("{orderNumber}")]
```

Comment:
Why do we currently do not have a `Route` attribute here?

Reply:
because it was not needed, given that `id` is the default input parameter name for actions as stated in the default API configuration:

```
config.Routes.MapHttpRoute(
    name: "DefaultApi",
    routeTemplate: "api/{controller}/{id}",
    defaults: new { id = RouteParameter.Optional }
);
```

In order to change from `id` to `orderNumber` to make it explicit we need to instruct the WebAPI routing engine on how to match this route that is basically new.